.gitignore

```diff
@@ -1,3 +1,2 @@
 .idea
-composer.lock
```

Comment:
> From a support perspective I prefer the latter, I don't want to know when a dependency breaks my build; I want to know when I break the build.

I have something to add to this point. If you test against the lock file which is ignored for other people you will not notice any break in some never version. This way the users of your library will notice and need to report it to you to fix it. If you always test against the latest version, you see breaking changes before they get reported to you and can react to them earlier.

Having the lock file means you test against a version combination that composer itself would never install (assuming there are newer versions out there already).